### PR TITLE
parted: proper fix for change of partition label case

### LINKED ISSUE #522: parted: unexpected partition table wipe
##### SUMMARY

If the disk label is unspecified and actual label is different from msdos, then parted module wipes the partition table.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

parted

##### ANSIBLE VERSION

```paste below
ansible 2.9.9
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/ei-grad/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.8/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.8.3 (default, May 17 2020, 18:15:42) [GCC 10.1.0]
```

##### CONFIGURATION

```paste below

```

##### STEPS TO REPRODUCE

Execute the following task on disk with gpt label:

```yaml
  - name: create partition for lvm on each nvme
    parted:
      device: "/dev/nvme1n1"
      number: 3
      part_start: 197257216s
      part_end: 100%
      state: present
```

##### EXPECTED RESULTS

New partition is created.

##### ACTUAL RESULTS

The partition table is wiped.

```paste below
failed: [myhostname] (item=/dev/nvme1n1) => changed=false 
  ansible_loop_var: item
  err: |-
    Error: Partition(s) 2 on /dev/nvme1n1 have been written, but we have been unable to inform the kernel of the change, probably because it/they are in use.  As a result, the old partition(s) will remain in use.  You should reboot now before making further changes.
  item: /dev/nvme1n1
  msg: 'Error while running parted script: /sbin/parted -s -m -a optimal /dev/nvme1n1 -- unit KiB mklabel msdos mkpart primary 197257216s 100%'

```

### PR DESCRIPTION
**Pull request by rosowiecki**
calling mkpart even when partition existed before mklabel call 
##### SUMMARY
Fixes #522

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME
parted
##### ADDITIONAL INFORMATION

Below playbook created empty partition label, even when partition table was not used by kernel.
My first diagnose of #522 was wrong, this is proper fix

```yaml
- hosts: localhost
  gather_facts: no
  become: yes
  tasks:
   - name: create msdos partition
     parted: 
       device: /dev/loop0
       number: 1
       state: present
       label: msdos

   - name: create gpt partition
     parted: 
       device: /dev/loop0
       number: 1
       state: present
       label: gpt
```

### CODE REVIEW

File: changelogs/fragments/522-parted_change_label.yml
Comment (felixfontein) [inline changelogs/fragments/522-parted_change_label.yml:2]:
```suggestion
 - "parted - fix creating partition when label is changed (https://github.com/ansible-collections/community.general/issues/522)."
```

**Comment by felixfontein** [comment]
bot_status

**Comment by Andersson007** [comment]
@ColOfAbRiX could you look at the changes please

**Comment by Andersson007** [comment]
i just emailed the author as well

**Comment by Andersson007** [comment]
closed-reopened to rerun tests

**Comment by Andersson007** [comment]
@ei-grad could you check this manually and confirm if this fixes the bug and doesn't break any current workflows?

**Comment by Andersson007** [comment]
cc @ColOfAbRiX

**Comment by Andersson007** [comment]
@e-grad @ColOfAbRiX are you around?

**Comment by Andersson007** [comment]
@ei-grad

**Comment by Andersson007** [comment]
@felixfontein thinks this PR is ok, if nobody complains we will merge this beginning of next week

**Review by felixfontein — APPROVED** [review]

**Review by Andersson007 — APPROVED** [review]
@rosowiecki thanks for the PR, we'll merge it next week if nobody is against. Thanks for your contribution!

**Comment by Andersson007** [comment]
@rosowiecki hi, i noticed that you did a lot of stuff for `parted` that is really cool!
Would you like to be added as the module's maintainer to .github/BOTMETA.yml?
Your shipit will be counted by ansible bot for authomerge of bugfixes and minor changes (needs 2) and you'll get notifications when related issues/prs appear (i don't think there will be a lot and nothing obligatory, of course). How about that?

**Comment by rosowiecki** [comment]
@Andersson007 Sure, count me in!

**Comment by Andersson007** [comment]
@rosowiecki cool! done https://github.com/ansible-collections/community.general/pull/1254, waiting for your approval, thanks!

**Comment by mnussbaum** [comment]
Trying to connect some dots here, should this resolve the old issue at https://github.com/ansible/ansible/issues/38594?

**Comment by ei-grad** [comment]
Both issues are not resolved. This RR is about "we just wiped partition table because user didn't specify label type [this is expected behaviour], but we steel need to create the new partition [failing to do so was not expected behaviour]".

**Pull request merged**
